### PR TITLE
[FIX] data/io.py Metadata file not saved anymore when it is empty

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
Do not save the metadata file (when saving the data) if this is empty (0 Bytes)

##### Description of changes
Change if statement and add third condition. It implies that attribute should not be empty. Add test file to check if metadata is saved when empty

##### Includes
- [x] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

File: Orange/tests/test_metadata_save.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+class TestMetaData(unittest.TestCase):
+    def setUp(self):
+        DiscreteVariable._clear_cache()
+
+    def test_metadata(self):
+        tempdir = tempfile.mkdtemp()
+        table = Table("iris")
+        fname = path.join(tempdir, "out.tab")
+        TabReader.write_table_metadata(fname, table)
+        self.assertEqual(path.isfile(fname + ".metadata"), True)
```
Comment (astaric) [inline Orange/tests/test_metadata_save.py]: You could use self.assertTrue

File: Orange/tests/test_metadata_save.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        shutil.rmtree(tempdir)
+
+    def test_no_metadata(self):
+        tempdir = tempfile.mkdtemp()
+        table = Table("titanic")
+        fname = path.join(tempdir, "out.tab")
+        TabReader.write_table_metadata(fname, table)
+        self.assertEqual(path.isfile(fname + ".metadata"), False)
```
Comment (astaric) [inline Orange/tests/test_metadata_save.py]: You could use self.assertFalse

File: Orange/data/io.py [hunk 3]
```diff
@@ -373,7 +373,7 @@ def write(cls, filename, data):
 
     @classmethod
     def write_table_metadata(cls, filename, data):
-        if isinstance(filename, str) and hasattr(data, 'attributes'):
+        if isinstance(filename, str) and hasattr(data, 'attributes') and data.attributes:
```
Comment (astaric) [inline Orange/data/io.py]:
```
hasattr(data, 'attributes') and data.attributes
```
could be simplified to
```
getattr(data, 'attributes', None)
```

File: Orange/tests/test_metadata_save.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# Test methods with long descriptive names can omit docstrings
```
Comment (astaric) [inline Orange/tests/test_metadata_save.py]: Is there a special reason this tests were not placed in `Orange/tests/test_tab_reader.py`? They only test TabReaders functionality and all other TabReader tests are in that file.

File: Orange/tests/test_metadata_save.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+from Orange.data import Table, DiscreteVariable
+from Orange.data.io import TabReader
+
+
+class TestMetaData(unittest.TestCase):
+    def setUp(self):
+        DiscreteVariable._clear_cache()
+
+    def test_metadata(self):
+        tempdir = tempfile.mkdtemp()
```
Comment (astaric) [inline Orange/tests/test_metadata_save.py]: This tests assume that iris has metadata and titanic has not. If this would change in the future (if we added metadata for titanic as well), the second test would fail and will need to be fixed, even though the change would have nothing to do with the saving of metadata.

I suggest that you manually set the attributes dict to whatever you need in a test and then do the testing. This way, attributes dict will be the one you expect it to be, even if the datasets change in the future.
